msdb/msdb/views/search.py

```diff
@@ -10,11 +11,86 @@
 def search(request):
     if request.method == "POST":
         search_string = request.POST.get("query")
+        queries = search_string.split()
+
+        # Find movies
+        movies = search_movie(queries)
 
-        movies_result = Movie.objects.filter(reduce(operator.and_, (Q(title__icontains=x) for x in search_string.split(' '))))[:10]
-        users_result = User.objects.filter(reduce(operator.and_, (Q(name__icontains=x) for x in search_string.split(' '))))[:10]
+        # Find people
+        people = search_people(queries, search_string)        
 
-        context = dict(query=search_string, movies=movies_result, users=users_result)
+        # Find users
+        users = search_users(queries)
+
+        context = dict(search_string=search_string, movies=movies, users=users, people=people)
         return render(request, "search/search.html", context)
     else:
         return redirect('home')
+
+
+def search_movie(queries : list):
+    movies_result = Movie.objects.filter(reduce(operator.and_, (Q(title__icontains=x) for x in queries)))[:10]
+    # if no movies are found, do the search with or operator instead
+    if movies_result.count() == 0:
+        movies_result = Movie.objects.filter(reduce(operator.or_, (Q(title__icontains=x) for x in queries)))[:10]
+
+    return movies_result
+
+
```

Comment:
again, as above, I would generally stay away from keeping 2 empty lines (point for discussion though)